cmd/esc/cli/env_run.go

```diff
@@ -127,19 +127,29 @@ func newEnvRunCmd(envcmd *envCommand) *cobra.Command {
 				return err
 			}
 
-			ref, args, err := envcmd.getExistingEnvRef(ctx, args)
-			if err != nil {
-				return err
+			// If -- was used but there's no arguments, it means no environment was specified before the command
+			if len(args) == 0 || cmd.ArgsLenAtDash() == 0 {
+				return fmt.Errorf("no environment specified")
 			}
 
-			if len(args) == 0 {
+			if len(args) == 1 {
 				return fmt.Errorf("no command specified")
 			}
-			command, err := envcmd.esc.exec.LookPath(args[0])
+
+			// First argument is always the environment name
+			envRefArgs := args[:1]
```

Comment:
Is there a reason to split the args like this, or is it just for clarity?? I expect the existing code here still works and doesn't need any changes

Reply:
Reverted back, Originally I was doing more parsing on the env name for improved error reporting but I think that's better to be handled on a different pr.